invoker/core/src/test/java/com/google/cloud/functions/invoker/http/HttpTest.java

```diff
@@ -271,44 +274,54 @@ public void multiPartRequest() throws Exception {
           assertThat(bytesPart.getFileName()).hasValue("/tmp/binary.x");
           assertThat(bytesPart.getContentLength()).isEqualTo(RANDOM_BYTES.length);
           assertThat(bytesPart.getContentType()).hasValue("application/octet-stream");
-          // We only see ["buh"] here, not ["baz", "buh"], apparently due to a Jetty bug.
-          // Repeated headers on multi-part content are not a big problem anyway.
           List<String> foos = bytesPart.getHeaders().get("foo");
-          assertThat(foos).contains("buh");
+          assertThat(foos).containsExactly("baz", "buh");
+
           byte[] bytes = new byte[RANDOM_BYTES.length];
           try (InputStream inputStream = bytesPart.getInputStream()) {
             assertThat(inputStream.read(bytes)).isEqualTo(bytes.length);
             assertThat(inputStream.read()).isEqualTo(-1);
             assertThat(bytes).isEqualTo(RANDOM_BYTES);
           }
         };
-    try (SimpleServer server = new SimpleServer(testServlet)) {
+    try (SimpleServer server = new SimpleServer(testHandler)) {
       testReference.set(test);
-      Request request = httpClient.POST(uri).header("foo", "oof").content(multiPart);
+      org.eclipse.jetty.client.Request request = httpClient.POST(uri)
+          .headers(m -> m.put("foo", "oof"))
+          .body(multiPart);
       ContentResponse response = request.send();
       assertThat(response.getStatus()).isEqualTo(HttpStatus.OK_200);
       throwIfNotNull(exceptionReference.get());
     }
   }
 
-  private static class HttpRequestServlet extends HttpServlet {
+  private static class HttpRequestHandler extends Handler.Abstract {
     private final AtomicReference<HttpRequestTest> testReference;
     private final AtomicReference<Throwable> exceptionReference;
 
-    private HttpRequestServlet(
+    private HttpRequestHandler(
         AtomicReference<HttpRequestTest> testReference,
         AtomicReference<Throwable> exceptionReference) {
       this.testReference = testReference;
       this.exceptionReference = exceptionReference;
     }
 
     @Override
-    protected void doPost(HttpServletRequest req, HttpServletResponse resp) {
+    public boolean handle(Request request, Response response, Callback callback) {
       try {
-        testReference.get().test(new HttpRequestImpl(req));
+        if (!HttpMethod.POST.is(request.getMethod())) {
```

Comment:
Can you add a comment to mention that this restriction is just the result of the migration from servlets and if a test case required this handler could be extended to accept GET requests.

Reply:
This was just a straight adaption of the servlet because it only overrided `doPost()`, but none of the tests are trying other HTTP methods and expecting a 405.

So I think its best I just remove this block of code.